\documentclass{article}
\usepackage{fullpage}
\usepackage{amsmath,amssymb,amsthm}
\usepackage{graphicx}
\usepackage{natbib}
\usepackage[hidelinks]{hyperref}
\usepackage[usenames,dvipsnames]{color}
\usepackage{authblk}
\usepackage[margin = 1.5in, includehead, includefoot, headsep=0.3in]{geometry}

\bibliographystyle{plainnat}

\usepackage{fancyhdr}
\pagestyle{fancy}
\fancyhead{}
\fancyfoot{}
\fancyhead[RO]{\small\thepage}
\fancyhead[LE]{\small\thepage}
\renewcommand{\headrulewidth}{0pt}

\newcommand{\rate}[2]{\text{rate}(#1\mapsto#2)}
\newcommand{\R}{\mathbb{R}}
\newcommand{\Z}{\mathbb{Z}}
\renewcommand{\P}{\mathbb{P}}
\newcommand{\E}{\mathbb{E}}
\DeclareMathOperator{\var}{var}
\DeclareMathOperator{\cov}{cov}
\newcommand{\deq}{\overset{\scriptscriptstyle{d}}{=}}
\DeclareMathOperator{\Pois}{Pois}
\newcommand{\given}{\,\vert\,}
\newcommand{\Given}{\,\bigg\vert\,}
\newcommand{\st}{\,\colon\,} % such that
\newcommand{\floor}[1]{{\left\lfloor #1 \right\rfloor }}
\newcommand{\one}{\mathbf{1}}
\newcommand{\aprod}[2]{{\langle{#1},{#2}\rangle}}
\newcommand{\grad}{\nabla}
\newcommand{\diffop}{\mathcal{L}}

\newtheorem{definition}{Definition}
\newtheorem{lemma}{Lemma}
\newtheorem{corollary}{Corollary}
\newtheorem{assumption}{Assumption}
\newtheorem{theorem}{Theorem}

\newcommand{\plr}[1]{{ \color{blue} #1}}
\newcommand{\details}[1]{\relax}

\title{Gene flow in patchy habitats}

\date{\normalsize\today}

\author[1]{Alison M.~Etheridge}
\author[2]{Peter L.~Ralph}
\author[1]{Aaron S.~A.~Smith}
\affil[1]{Department of Statistics, University of Oxford}
\affil[2]{Institute for Ecology and Evolution, Departments of Mathematics and Biology, University of Oregon}

%%%%% Running Title %%%%%%%%%%%%%%%%%%%%%%%

\fancyhead[CO]{\MakeUppercase{\small{Gene flow in patchy habitats}}}

%%%%% Running Authors %%%%%%%%%%%%%%%%%%%%%

\fancyhead[CE]{\MakeUppercase{\small{A.~M.~Etheridge, P.~L.~Ralph, A.~S.~A.~Smith}}}



\begin{document}

\maketitle

\textbf{Title brainstorm space}
Note: need to make the title not just about big/small.
\begin{itemize}
    \item Life in a big-small world
    \item Gene flow and boom-bust populations
    \item Tracking ancestry in patchy populations
    \item Genealogies in transient population explosions
    \item Genealogies of transient populations
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstract}
    The demographic details of an individual-based stochastic population model
    determine not only the time evolution of population sizes,
    but also the dynamics of \emph{lineages} --
    the description of where the ancestors of modern individuals lived at times in the past.
    Here, we formulate several simple models of population dynamics
    in which the motion of lineages traced back through time
    can be explicitly described,
    and highlight the role of the spatial and temporal autocorrelation structure
    in determining long-term behavior of lineage movement.
    The typical displacement between an individual
    and their ancestor at some time in the past
    may differ substantially from that predicted 
    based on the typical displacement between parent and child.
    SAY SOMETHING MORE CONCRETE ABOUT THE MODELS
\end{abstract}

\setcounter{tocdepth}{2}
\tableofcontents

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Introduction}

The dynamics of a population 
are determined by the demography of the individuals that make up the population:
fecundities, mortalities, movement rates, and so forth.
Stochastic population models can be lead to complex behavior,
especially across continuous geography,
and a great deal of work has focused on describing them (CITE).
Inferring these demographic parameters of populations
can be of substantial practical importance,
as for instance when working to conserve an endangered species
or to control an invasive species or disease vector.
Although reproduction is a fundamental way by which populations grow,
the genealogical connections between parents and offspring can be ignored
if one is only interested in population size changes.
However, these relationships clearly contain important information about population demography,
and so there is substantial interest in using the patterns of relatedness inferred from genetic data
to learn about the demographic dynamics and history of real populations.
For instance, genome sequencing has provided surprising information about the history of our own species
both in the recent and distance past (CITE).
However, the relationship between population demographic dynamics
and the resulting patterns of genealogical relationships have only been described
in the simplest possible models.
In this paper, we study how genealogical inheritance, traced back through time,
is determined by the (forwards-time) process of population size changes.
The models we study are greatly simplified,
but are chosen to highlight the important aspects of this relationship.

If we begin with a single individual from some population alive today 
and choose a particular bit of her genome,
then, looking back through history, 
we can imagine recording the geographic location of the ancestor from whom that individual
inherited that bit of genome.
This spatial motion, as a function of amount of time in the past,
we call a \emph{lineage}.
For concreteness, suppse we follow inheritance of the chloroplast,
and thus the maternal lineage, in a population of plants.
Suppose we know the birth location of a given ancestor at a particular time.
Roughly speaking, the location of the next ancestor might be chosen from among the possible ancestors,
weighted by the reproductive output of those ancestors that established at that location
at that time.
On a broad scale, therefore,
lineages when traced by through time should tend to follow regions of high population density,
if the regions have correspondingly high total reproductive output.

The \emph{dispersal distance}, $\sigma$, characterizes the typical distance between parent and offspring
(concretely, it is the standard deviation of the displacement between the two along an arbitrary axis).
Imagining a lineage as a random walk, one might guess that in a homogeneous population, 
the distance from a given individual to their ancestor from $T$ generations ago
is proportional to $\sigma \sqrt{T}$.
Although this may be close in some cases, this can be dramatically wrong in others.
The \emph{effective dispersal distance}
is the corresponding quantity seen by averaging back along a lineage --
if $D_t$ is the distance from an individual to their ancestor living $t$ generations ago,
then $\var[D_t] \approx t \sigma_e^2$.
For instance,
if there is significant competition between siblings, 
individuals that disperse further from their parents may be more successful.
Since these long dispersers are more successful, they contribute more to the modern-day gene pool,
and hence are more likely to be seen when looking back along a lineage --
so, in this case, we would expect $\sigma_e > \sigma$.
Conversely, if habitat quality is highly heterogeneous,
so that longer-distance dispersers are likely to end up in an inhospitable region,
then $\sigma_e$ may be substantially less than $\sigma$.
More generally, spatial and temporal autocorrelation in habitat quality
can lead to correlations between dispersal ability and long-term fitness,
and therefore selection on dispersal ability itself (CITE).

In this paper, we study the movement of linages in populations whose sizes
are entirely determined by exogenous factors, e.g., transient, patchy resources.
By varying the spatial and temporal autocorrelation of habitat quality in several models,
we can highlight how these determine the long-term behavior of lineage movement,
and hence spatial patterns of relatedness and gene flow.
For a concrete situation, imagine that we are tracking populations of a soil microbe
across a grassland, and the microbe is tightly associated with a particular type of shrub.
A low density of microbes are found in the soil away from shrubs,
but much higher densities live in the root zones of patches of shrubs
that expand, contract, appear, and dissappear over time.
New patches of shrubs initiate only rarely, but when they do,
the lucky microbes living nearby to the location where the patch originated reproduce rapidly, 
and since the area surrounding the patch produces very few offspring,
the microbes found in that patch are descdendants of those initially lucky individuals
for as long as the patch exists.
This means that as one traces back the lineage of a microbe found in a patch today,
the lineage must remain in that patch until its origination --
and so, the motion of lineages is strongly constrained by the dynamics of the patches.

Although it is not true in general,
we put in place assumptions that make the location of a lineage,
viewed moving back through time, a Markov process.
Since the structure of the habitat is random,
we are studying a random walk in a random environment,
which XXX is something other people study too CITE.

NOTE: discuss \citet{birkner2013directed} and \citet{birkner2016random} somewhere.

\subsection*{The basic model}

We begin with some calculations to motivate the model.
Suppose that we study a species of plant that lives in discrete populations,
and that every year, the entire population reproduces and then dies.
To make things even simpler, we suppose that every individual makes a large number of seeds,
-- the same number regardless of location --
and that we only follow ancestry to the seed parent (ignoring pollen).
Then, if the proportion of the seeds of a plant at location $x$
that arrive at location $y$ is $m(x,y)$,
and the number of individuals at location $x$ in the $k^\text{th}$ generation ago is $N(x,k)$,
then the total number of seeds from $x$ that arrived at $y$ in the subsequent generation
is proportional to $N(x,k) m(x,y)$.
If all these seeds are roughly equivalent, so that the subsequent generation was composed of a random choice
of all available seeds, then the chance that a given plant living at $y$ some $k-1$ generations ago
grew from a seed originating in $x$ in the previous generation, is
\begin{align} \label{eqn:basic_movement}
    \frac{N(x,k) m(x,y)}{\sum_z N(z,k) m(z,y)},
\end{align}
where the sum is over all locations.

In this paper,
we then assume that the lineage of an individual can be described as a Markov process given the population sizes --
concretely, if $L_k$ is the location $k$ generations ago 
of the (maternal) ancestor of a given individual who is today randomly chosen from $L(0)$,
then given the population sizes histories $(N(z,k))_{z,k}$,
the series of locations $(L_k)_{k \ge 0}$ forms a Markov chain with 
$\P\{L_{k-1} = y \given L_k = x\}$ given by equation \eqref{eqn:basic_movement}.
This is a standard assumption \citep{wakeley2009coalescent},
and generally requires all population sizes to be large.

We then work in two situations: XXX continuous, and patchy XXX.
Furthermore, we will assume that there are only two types of habitat, ``bad'' and ``good'',
and that population sizes in ``good'' habitat are so large that, given the opportunity,
lineages will always move from bad to good habitat.
In the motivating example above, good habitat occurs where the shrub lives.
Our analysis will occur mostly in one spatial dimension,
beginning on the discrete lattice, with nearest-neighbor dispersal.
We will assume that the habitat evolves according to a number of different Markov processes,
each with the same stationary distribution (described below).
Taking the environmental dynamics to be reversible and at stationarity,
we can then simply track the dynamics of the lineage and the habitat together
back into the remote past.
In spite of its apparent simplicity,
the model allows us to describe a number of important features of the general problem,
while retaining some analytical and numerical tractability.


The structure of the paper is as follows OR SOME TRANSITION HERE.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Diffusion approximations}

Consider a population with more or less continuous density,
and write the number of individuals per unit area around location $x$ as $\rho(x)$.
Suppose that in each unit of time, each individual at $x$ 
gives birth to an average of $\gamma(x)$ juvenile offspring,
and then either dies (with probability $\mu(x)$) or survives.
These juveniles each disperse to a nearby location $y$
chosen with probability proportional to $q(y-x)$,
and then ``establish'' there with probability $r(y)$ (otherwise, they die).
The time dynamics of the population density $\rho$
are determined by the fecundity $\gamma$, the dispersal distribution $q$,
the recruitment rates $r$, and the death rates $\mu$.
In general, we expect these rates to depend on the state of the population
-- for instance, the local recruitment rate $r(x, t)$ may be a decreasing function
of the population density $\rho(x, t)$.
Of course, individuals are discrete, and we work in continuous space,
so what exactly is the ``population density''?
Our goal here is a heuristic derivation, so we'll ignore this,
and move on to the question we are really interested in:
how a lineage moves across space when looking back through time in this population.

To answer this, we need to describe how likely a lineage is to move from location $y$ to location $x$.
What is the probability that a randomly sampled individual from $y$
has just been born to a parent who lived at $x$?
At first, just consider the maternal parent.
The net change in population density at $y$ between times $t - dt$ and $t$ is
\begin{align*}
    \rho(y, t) - \rho(y, t - dt)
    \approx
    \left\{
        \int \rho(x, t) \gamma(x, t) q(y - x) r(y, t) dx - \mu(y, t) \rho(y, t) 
    \right\} dt .
\end{align*}
The first term is the new individuals,
so the probability that our randomly chosen individual at $y$ is newly established is
this quantity divided by the local population size.
Partitioning these newly establishing individuals,
the probability that an individual at $y$ is newly established
with a maternal parent in a unit of area $dx$ near $x$ at time $t$ is
\begin{align} \label{eqn:jump_intro}
    \frac{
        \rho(x, t) \gamma(x, t) q(y - x) r(y, t)
    }{ \rho(y, t) } dx dt .
\end{align}

Next, we'll assume that the population density and related quantities
change over a larger scale than dispersal.
Let $R = (R_1, R_2)$ denote the random displacement from maternal parent to child
in the two coordinate axes,
so that $\P\{R = y\} = q(y)$,
and suppose that the mean displacement is $\E[R] = m$.
(Since we are in two dimensions, $m = (m_1, m_2)$ is a vector.)
For simplicity, we assume displacements along the two coordinate axes 
are uncorrelated and have equal variance,
so that $\cov[R] = \sigma^2 I$, where $I$ is the identity matrix.
Then, if $m$ and $\sigma^2$ are both small (and of the same order),
using a Taylor series,
\begin{align} \label{eqn:taylor}
    \int f(x) q(y - x) dx % &= \int f(y - r) q(r) dr \\
    &= \E[f(y - R)] \\  % &= \E[f(y) - \E[R] \grad f(y) +  \ldots \\
    &\approx f(y) - m \cdot \grad f(y) + \sigma^2 \Delta f(y) ,
\end{align}
where $m \cdot \grad f(y) = \sum_i m_i \partial_{y_i} f(y)$
and $\Delta f(y) = \sum_i \partial^2_{y_i} f(y)$.
\details{(There is also a term $\sum_{ij} m_i m_j \partial_{ij} f(y)$, but it is second-order in $m$.)}

If a lineage is at location $y$ at time $t$,
then the mean displacement of the lineage at time $t - dt$ relative to $y$
is the integral of equation \eqref{eqn:jump_intro} against $x - y$.
Using the Taylor approximation \eqref{eqn:taylor} with $f(x) = (x - y) \rho(x,t) \gamma(x,t)$,
this is
\begin{align*}
    \E[dL_t]
    &=
    \frac{dt}{\rho(y, t)} \int (x - y) \rho(x, t) \gamma(x, t) q(y - x) r(y, t) dx \\
    &\approx
    \frac{r(y, t)}{\rho(y, t)} \left\{
            - m \left( \rho \gamma \right)(x, t)
            + \sigma^2 \grad\left( \rho \gamma \right)(y, t)
        \right\} dt \\
    &=
    r(y, t) \gamma(y, t) \left\{
        \sigma^2 \grad \log\left(\rho \gamma \right)(y, t) 
        - m 
    \right\} dt,
\end{align*}
where $\grad(\rho \gamma)(y,t)$ is the gradient of the product of $\rho$ and $\gamma$,
i.e., the gradient of the total production of offspring.
Similarly,
the variance of the displacement of the lineage is
\begin{align*}
    \E[dL_t^2]
    &=
        \frac{dt}{\rho(y, t)} \int (x - y)^2 \rho(x, t) \gamma(x, t) q(y - x) r(y, t) dx \\
    &\approx 
        % \frac{dt}{\rho(y, t)} \frac{\sigma^2}{2} \rho(y, t) \gamma(y, t) r(y, t) \\ &=
        \sigma^2 r(y, t) \gamma(y,t) dt .
\end{align*}

These describe the mean and variance of a time-inhomogeneous random walk across the landscape,
that is driven by the population dynamics.
If the displacements across many generations are sufficiently independent,
then $L_t$ is well-approximated by the diffusion that satisfies
\begin{align}
    dL_t =  r(L_t) \gamma(L_t)
        \left\{
            \left( 2 \sigma^2 \grad \log \left(\rho \gamma\right)(L_t) - m \right) dt
            + \sigma dB_t
        \right\} .
\end{align}
In words, this says that $L_t$ behaves like a Brownian motion
run at speed $\sigma r(y) \gamma(y)$ when it is at location $y$,
in the potential
\begin{align}
    V(y) = \rho(y) \gamma(y) e^{-my/(2\sigma^2)} ,
\end{align}
which has stationary distribution
\begin{align}
    \pi(y) = \frac{\rho(y)}{r(y)} e^{-my/(2\sigma^2)} .
\end{align}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Discrete, patchy habitats}

We begin with a random habitat on the one-dimensional lattice,
with ``bad'' habitats encoded as 0 and ``good'' habitats encoded as 1.
The spatial distribution of habitat at any particular time can be thought of as
a sequence of ``patches'' of consective runs of 1s, separated by runs of at least one 0.
We will model these by saying that each patch has a Geometric($q$) number of 1s,
each space between two patches is composed of a Geometric($p$) number of 0s,
and that all such are independent.
In other words, the probability that the habitat at location zero is 1 is $p/(p+q)$,
and the sequence of habitats one encounters as one walks along the lattice (in either direction)
is a Markov chain that moves $1 \to 0$ with probability $q$ and moves $0 \to 1$ with probability $p$.
(Note that in contrast to convention in some fields, $p$ is \emph{not} assumed to be $1-q$.)
We will also specify the temporal dynamics of the habitat in a way that preserves this distribution.

The habitat is then described by a binary sequence $\omega \in \{0,1\}^\Z$,
and we call the probability distribution above $\pi$.
An equivalent description of $\pi$ is as a \emph{random cluster measure}, 
also sometimes known as a Fortuin-Kasteleyn measure CITE.
If $(e_0, e_1, \ldots, e_L)$ is a possible subsequence of habitats,
with $e_0 = e_L = 0$ for definiteness,
then the probability under $\pi$ that $\omega_{k+i} = e_i$ for all $0 \le i \le L$
is
% (q/(p+q)) (1-p)^(N0-Nc) (1-q)^(N1-Nc) p^Nc q^Nc
\begin{align}
    (1-p)^{N_0} (1-q)^{N_1} \left(\frac{pq}{(1-p)(1-q)}\right)^{N_c},
\end{align}
where $N_0$ and $N_1$ are the number of 0s and 1s, respectively,
and $N_c$ is the number of ``patches'' of good habitat (i.e., the number of runs of 1s).

\subsection{No temporal correlations in habitat}
The simplest case is when the habitat is simply resampled independently from this distribution
every generation,
and that offspring are divided evenly between the location of their parent and the locations
directly on either side of the parent.

\begin{lemma}
    Suppose that the habitat is resampled independently in each generation,
    and that $m(x,x) = m(x,x+1) = m(x,x-1) = 1/3$.
    The variance of the displacement of an ancestral lineage over a single generation is 
    $$\begin{aligned}
        \var[L_1] = \frac{2}{3} + \frac{pq}{3(p+q)}\left(1 - (p+q)\right) .
    \end{aligned}$$
    In particular, lineages move faster than those in a homogeneous environment
    if and only if $p+q < 1$.
\end{lemma}

The (elementary) proof is XXX LATER.

In other words, 
more positive spatial correlations in the environment cause the walk to move more quickly, 
and more negative ones (ones more in the negative direction) 
cause it to move more slowly. 
The eigenvalues of the transition matrix of the environment 1 and $1 - (p + q)$, 
so the transition corresponds to that from a smooth to an oscillating habitat.
At first this result seems a little counterintuitive. 
However, consider what happens when $p = q = 1$, 
so that the habitat alternates between `good' and `bad' in space. 
Then in each generation, 
there is probability $1/2$ that the ancestral lineage does not move at all, 
and probability $1/2$ that it moves to a neighbouring deme. 
The variance of the movement of a lineage over one generation is then $1/2$, 
less than the $2/3$ corresponding to the symmetric walk corresponding to $m$, 
which chooses each of the three potential ancestral demes with equal probability.
Conversely, if $p$ and $q$ are both small, then there is some chance a lineage will find itself
near the edge of a large patch of good habitat, thus giving it an extra ``push''.
However, the speed is maximized at $p=q=1/4$, but is only sped up by a factor of $33/32$:
the effect is not large.

%%%%%%%%%%%% double-check that:
% > pq <- expand.grid(p=seq(0,1,length.out=41),q=seq(0,1,length.out=41))
% > f <- function (p,q) 2/3 + p*q*(1-p-q)/(3*(p+q))
% > pq$z <- with(pq, f(p,q))
% > f(1/4,1/4)
% [1] 0.6875
% > max(pq$z,na.rm=TRUE)
% [1] 0.6875
%%%%%%%%%%%


It is also interesting to note that since the lineage only moves to bad habitat
if there is no available good habitat,
the proportion of the time a lineage spends in bad habitat is $q(1-p)^2/(p+q)$.
This is the proportion of future generations who are direct descendants of today's bad habitats,
i.e., proportional to the total long-term fitness of bad habitats.
This is despite the fact that in any particular generation, the vast majority of individuals
may live in good habitats --
but, long-term genetic contributions to the population
is mostly determined by who initiates new patches, rather than who lives in them.

\subsection{Different intergeneration times for the habitat and the resident population}

Of course, 
there is no reason to suppose that the habitat and the resident population 
have the same intergeneration time. 
It is instructive to consider what happens if we suppose that several generations of the population
pass during each time step of the habitat.
Evidently, if the habitat never changes, 
then any lineage will become ``trapped'' in a cluster of good habitat.

It is tedious, but not difficult, to prove the following lemma.
\begin{lemma}
    Suppose that the habitat is resampled every two generations 
    (in between which it is held fixed). 
    The variance of the position of a lineage after two generations is
    \begin{align} \label{eqn:two_step}
        \var[L_2] 
            = \frac{4}{3} + 
            \frac{pq}{18(p+q)} \left\{
                47 - 77p - 62q + 27pq + 28p^2 + 7q^2
            \right\} .
    \end{align}
\end{lemma}

The term $4/3$ in \eqref{eqn:two_step} 
is precisely what we would have for the symmetric walk in a homogeneous environment. 
If $p$ and $q$ are close to one then the second term is always negative, 
and the walk slows down in the random environment.
If $p$ and $q$ are close to zero, 
then the walk always goes faster in the random than in the homogeneous environment, 
but the factor $pq$ tells us that the effect is small.

It is not hard to convince oneself that the longer the habitat is held fixed, 
the smaller $p$ and $q$ must be 
if ancestral lineages are not to be slowed down compared to simple random walk. 
We require $q$ to be small in order that a cluster is big enough 
that the motion of a lineage is not be too constrained before the habitat is resampled;
we require $p$ to be small so that we rarely see clusters of favrouable environment 
and so donâ€™t become trapped.

The longer the habitat is held constant, 
(where time is measured in numbers of generations of the population), 
the more the ancestral lineages will be slowed compared to those sampled from a homogeneous population. 
Of course, holding the habitat constant and then resampling is not particularly natural, 
and so we now turn to finding a Markovian evolution of the environment. 
It is convenient to first recast our stationary distribution as a random cluster measure.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Changing habitats}

Holding the habitat constant for a fixed number of generations and then (independently) resampling, 
seems very artificial. 
Instead we should like it to evolve according to a Markov process. 
The simplest way to achieve this is to work in continuous time.

We allow each deme to change state at an instantaneous rate that is chosen in such a way
as to ensure that the detailed balance equations are satisfied, 
so that the dynamics of the habitat is reversible, 
and the distribution $\pi$ is stationary for the process. 
There are many ways to do this,
and part of our purpose is to compare 
the effects on the motion of ancestral lineages of different dynamics of the habitat, 
for which the underlying (stationary) distribution of the habitat is the same.

To maintain detailed balance, we need to set things up so that
the ratio of instantaneous transition rates between two configurations
is equal to the ratio of their probabilities under $\pi$.
The simplest way to do this is to flip one position at a time,
with an instantaneous rate that depends only upon the states of its two nearest neighbours. 
Concretely, if $\omega = (\ldots, \omega_{-1}, \omega_0, \omega_1, \ldots)$
is the current state of the process, then $\omega_0$ will flip
at a rate depending only on the triple $(\omega_{-1}, \omega_0, \omega_1)$.
We must therefore specify rates for the four pairs: 
$000 \leftrightarrow 010$,
$001 \leftrightarrow 011$,
$100 \leftrightarrow 110$,
and $101 \leftrightarrow 111$.

The first pair of rates gives us the instantaneous ates of
 ``births'' and ``deaths'' of patches of good habitat.
For detailed balance, we need the rates to satisfy
\begin{align*}
    \frac{\rate{000}{010}}{\rate{010}{000}}
    =
    \frac{\pi(\cdots010\cdots)}{\pi(\cdots000\cdots)}
    =
    \frac{pq}{(1-p)^2} .
\end{align*}
To do this, we define
\begin{align}
    \rate{000}{010} &= \gamma_b \frac{pq}{1-p}, \quad \text{and}\\
    \rate{010}{000} &= \gamma_b (1-p) .
\end{align}

The second and third pair determine how fast the left and right boundaries of a patch move, respectively.
It seems most parsimonius for these two to be symmetric,
so we will set $\rate{001}{011} = \rate{100}{110}$ and $\rate{011}{001} = \rate{110}{100}$.
Here, we need
\begin{align*}
    \frac{\rate{001}{011}}{\rate{011}{001}}
    =
    \frac{\pi(\cdots011\cdots)}{\pi(\cdots001\cdots)}
    =
    \frac{1-q}{1-p} ,
\end{align*}
so similar to before, we let
\begin{align}
    \rate{001}{011} &= \rate{100}{110} = \gamma_m (1-q), \quad \text{and} \\
    \rate{011}{001} &= \rate{110}{100} = \gamma_m (1-p) .
\end{align}

The remaining pair of rates, at which $111 \leftrightarrow 101$,
determines how fast patches split and merge.
Although this is plausible, allowing split/merge events will greatly complicate the analysis,
so as a final assumption we set these rates equal to zero.




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Discrete habitats and rapid turnover}

Proofs of whatever is left here.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Rescaling to continuous landscapes}

Proofs of whatever is left here.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Discussion}



\appendix


%%%%%%%%%%%%%%%%%%%%%%%
\section{Diffusion approximation calculations}

Here we will work through in more detail the arguments that lead to the diffusion approximation
for a lineage in a continuous landscape.
We still assume that the population has a continuous density,
without examining in detail what that means or how it is determined by the vital rates.

Recall that the rate of influx of establishing individuals from $x$ into $y$ at time $t$ is
$\rho(x) \gamma(x) q(y - x) r(y)$,
and let $R$ be a random seed dispersal displacement (i.e., $\P\{R = r\} = q(r)$),
For the moment, ignore the male gametes (so, we only follow a maternally inherited lineage).
Write $L_s$ for the location of the lineage $s$ units of time ago,
and suppose that $L_0 = y$.
Then, for a function $g( )$,
\begin{align*}
    \E[g(L_{dt}) \given L_0 = y]
    &\approx
        g(y) 
        + \frac{dt}{\rho(y)} \int \rho(x) \gamma(x) q(y - x) r(y) \left(g(x) - g(y)\right) dx \\
    &=
        g(y) 
        + dt \frac{r(y)}{\rho(y)} \E\left[ \rho(y - R) \gamma(y - R) \left(g(y - R) - g(y)\right) \right] .
\end{align*}
Now suppose that the mean and variance of $R$ are of the same small order:
$\E[R] = m$ and $\cov[R] = C$,
with $m_i$ and $C_{ij}$ small and of similar size for each $i$ and $j$.
Then, since $\E[R_i R_j] = \E[R_i] \E[R_j] + \cov[R_i, R_j] = (m_i m_j + C_{ij})$,
a Taylor expansion gives
\begin{align*}
    \E[f(R) - f(0)]
    &\approx
        \sum_i \E[R_i] \partial_{x_i} f(0) 
            + \sum_{ij} \E[R_i R_j] \partial_{x_i} \partial_{x_j} f(0) \\
    &=
        \sum_i m_i \partial_{x_i} f(0) 
            + \sum_{ij} \left(C_{ij} + m_i m_j\right) \partial_{x_i} \partial_{x_j} f(0) \\
    &=
        \sum_i m_i \partial_{x_i} f(0) 
            + \sum_{ij} C_{ij} \partial_{x_i} \partial_{x_j} f(0) + O(|m|^2) .
\end{align*}
To save on writing subscripts, let's give a name to this differential operator:
\begin{align*}
    \diffop(m, C) f(x) 
    :=
        \sum_i m_i \partial_{x_i} f(x) 
            + \sum_{ij} C_{ij} \partial_{x_i} \partial_{x_j} f(x) .
\end{align*}
Let's apply this to the expression above,
with $f(r) = \rho(y - r) \gamma(y - r) (g(y - r) - g(y))$.
First note that
\begin{align*}
    \partial_{x_i} f(0)
    &= - \rho(y) \gamma(y) \partial_{x_i} g(y),
\end{align*}
and that
\begin{align*}
    \partial_{x_i} \partial_{x_j} f(0)
    &= \partial_{x_i} \left(\rho(y) \gamma(y)\right) \partial_{x_j} g(y)
       + \partial_{x_j} \left(\rho(y) \gamma(y)\right) \partial_{x_i} g(y)
       + \left(\rho(y) \gamma(y)\right) \partial_{x_i} \partial_{x_j} g(y) .
\end{align*}
Therefore, since $C$ is symmetric,
\begin{align*}
    &
     \E\left[ \rho(y - R) \gamma(y - R) r(y) \left(g(y - R) - g(y)\right) \right]   \\
    &\qquad
     \approx
         - \rho(y) \gamma(y) \sum_i m_i \partial_{x_i} g(y) 
         + \rho(y) \gamma(y) \sum_{ij} C_{ij} \partial_{x_i} \partial_{x_j} g(y)
         + 2 \sum_{ij} C_{ij} \partial_{x_i} \left(\rho(y) \gamma(y)\right) \partial_{x_j} g(y)
     \\
     &\qquad
     = \rho(y) \gamma(y) \left\{ \diffop(-m,C) g(y) + 2 \grad(\log \rho(y) \gamma(y))^T C \grad g(y) \right\} .
\end{align*}

Putting this together,
the motion of a maternal lineage is described by
\begin{align*}
    \E[g(L_{dt}) \given L_0 = y]
    &\approx
    g(y) + r(y) \gamma(y)
    \left\{ 
        \diffop(-m,C) g(y)
        + 2 \grad(\log \rho(y) \gamma(y))^T C \grad g(y)
    \right\} .
\end{align*}
This is what we'd expect from a random walk directed by $-R$,
but with an additional ``push''
in the direction of twice the gradient of $\log(\rho(y) \gamma(y))$,
multiplied by the dispersal covariance matrix $C$.
In other words, the generator for a maternal lineage is approximately
\begin{align} \label{eqn:mat_generator}
    r \gamma \diffop\left(- m + 2 C \grad(\log \rho \gamma), C \right) .
\end{align}

What about male gametes, i.e., pollen?
Suppose the rate of production of pollen at $z$ $\rho(z) \delta(z)$,
and that the mean and covariance of pollen dispersal is $m^{(p)}$ and $C^{(p)}$ respectively.
Applying the calculations above after replacing $\gamma$ by $\delta$ and setting $r = 1$,
we see that a lineage that \emph{only} followed pollen movement
(if for instance seeds traveled only very small distances)
would have generator
\begin{align*}
    \delta \diffop\left(-m^{(p)} + 2  C^{(p)} \grad(\log \rho \delta), C^{(p)} \right) .
\end{align*}

Now, the lineage at an autosomal locus will step from offspring to mother every generation,
and then from mother to father half the time.
Trotter's formula lets us avoid doing the nasty double Taylor series,
and says that the combination of these two results in a generator of the form
\begin{align} \label{eqn:final_generator}
    \diffop\left(
        - r \gamma m - \frac{1}{2} \delta m^{(p)} 
        + 2 r \gamma C \grad(\log \rho \gamma)
        + \delta C^{(p)} \grad(\log \rho \delta)
        ,
        C + \frac{1}{2} C^{(p)}
    \right) .
\end{align}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Stationary distributions for reversible lineages}

In some cases, the process whose generator is described in \eqref{eqn:final_generator}
is reversible, which allows us to find its stationary distribution.
Let's work out when this happens:
suppose that $C$ is a (fixed) symmetric matrix and $a(x)$ is a function,
and consider the following symmetric form
(for functions $f$ and $g$ such that the boundary terms dissappear):
\begin{align*}
    \langle f, g \rangle
    &:=
    \int a(x) \sum_{ij} C_{ij} \partial_{x_i} f(x) \partial_{x_j} g(x) dx \\
    &=
    - \int f(x) \sum_{ij} C_{ij} \partial_{x_i} \left( a(x) \partial_{x_j} g(x) \right) dx \\
    &=
    - \int f(x) \sum_{ij} C_{ij} \left(
        \left[\partial_{x_i} a(x)\right] \left[\partial_{x_j} g(x)\right]
        + a(x) \partial_{x_i} \partial_{x_j} g(x) \right) dx \\
    &=
    - \int f(x) \diffop\left( C \grad a(x), a(x) C \right) g(x) dx \\
    &=
    - \int f(x) a(x) \diffop\left( C \grad \log a(x), C \right) g(x) dx .
\end{align*}
(Note that if $C$ varies across space then there is another term.)
This implies that if a process has generator
$\mathcal{G} = s(x) \diffop(C \grad \log a(x), C)$,
then letting $\pi(x) = a(x) / s(x)$,
\begin{align*}
    \int \pi(x) f(x) \mathcal{G} g(x) dx
    &=
        \int f(x) a(x) \diffop\left( C \grad \log a(x), C \right) dx \\
    &=
        \langle f, g \rangle ,
\end{align*}
which is symmetric, and so the process is reversible with respect to $\pi$.

For instance, if $m$ is constant,
generator \eqref{eqn:mat_generator} is of this form,
and is reversible with respect to
\begin{align}
    \pi(x) &= \frac{\rho(x) \gamma(x) e^{-m^T C^{-1} x / 2}}{ r(x) \gamma(x) } \\
        &= \frac{\rho(x)}{r(x)} e^{-m^T C^{-1} x / 2} .
\end{align}

Similarly, if $h$ is the elevation and $m(x) = \alpha \grad h(x)$
(e.g., if $\alpha < 0$ then dispersal is downhill),
and $C = \sigma^2 I$,
then the stationary distribution is
\begin{align}
    \pi(x) &= \frac{\rho(x)}{r(x)} e^{-\alpha h(x) / \sigma^2} .
\end{align}

%%%%%%%%%%%%%%%%%%%%%%%
\section{Proof of X}

%%%%%%%%%%%%%%%%%%%%%%%
\section{Proof of Y}


\nocite{*}
\bibliography{refs}


\end{document}
